library/private/camera_impl.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * @class   camera_impl
+ * @brief   A concrete implementation of camera
+ *
+ * A concrete implementation of camera that can provide a camera to control in a window
+ * It also defined implementation only API used by the libf3d.
+ * See camera.h for the class documentation
+ */
+
+#ifndef f3d_camera_impl_h
+#define f3d_camera_impl_h
+
+#include "camera.h"
+
+#include <memory>
+
+class vtkRenderer;
+class vtkCamera;
+namespace f3d::detail
+{
+class camera_impl : public camera
+{
+public:
+  /**
+   * Default constructor
+   */
+  camera_impl();
+
+  /**
+   * Default destructor
+   */
+  ~camera_impl() override;
```

Comment:
Rule of three, add a copy constructor, and a copy assignment operator. Ideally, it should even be rule of five and add move constructor/assignment.

Reply:
nevermind, the copy is deleted in the base class.